src/ngraph/ngraph_imperative.cc

```diff
@@ -87,69 +87,128 @@ void NGImperative::parse_ngraph() {
     }
 }
 
+bool compute_forward_imperative(const nnvm::NodeAttrs &attrs,
+                                const mxnet::OpContext &ctx,
+                                const std::vector<mxnet::NDArray> &inputs,
+                                const std::vector<mxnet::OpReqType> &req,
+                                const std::vector<mxnet::NDArray> &outputs) {
+  std::shared_ptr<Graph> op_ng;
+  int mode = static_cast<int>(GraphExeMode::kInfer);
+  if (!sparse_check(inputs) && !sparse_check(outputs)) {
+    // thread local cache for ngraph op
+    static thread_local NGIOpCache ngicache;
```

Comment:
Future maintainers might benefit from having the comment explain why a thread-local cache was used here.  I.e....

Was it to get caching without the potential slowdown or code-complexity of using a thread-safe cache?  If so, is this based on speculation or actual benchmarking?

Or is there something about thread affinities that make a thread-local cache especially likely to get a high cache-hit ratio?

Reply:
will clarify more, this is primarily done to safe use the cache object. No specific performance reasons.